### PR TITLE
Add CBMC proof of processValidFileContext

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of processValidFileContext function

Description
-----------
This PR adds files required for the CBMC proof of processValidFileContext function

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is formatted using Uncrustify.
- [ ] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/source/stubs.c [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    return err;
+}
+
+OtaOsStatus_t startTimerStub( OtaTimerId_t otaTimerId,
+                              const char * const pTimerName,
+                              const uint32_t timeout,
+                              OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:48]: Status should be initialized.

File: test/cbmc/source/stubs.c [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    __CPROVER_assert( pTimerName != NULL, "Error: TimerName cannot be NULL" );
+
+    __CPROVER_assert( callback != NULL, "Error: callback should be pointing to a function." );
+
+    return status;
+}
+
+OtaOsStatus_t stopTimerStub( OtaTimerId_t otaTimerId )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:62]: Status should be initialized.

File: test/cbmc/source/stubs.c [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    __CPROVER_assume( ( state >= OtaPalImageStateUnknown ) && ( state <= OtaPalImageStateInvalid ) );
+
+    __CPROVER_assert( pFileContext != NULL,
+                      "Error: pFileContext in the otaAgent is statically initialized and hence cannot be NULL." );
+
+    return state;
+}
+
+void otaAppCallback( OtaJobEvent_t eEvent,
+                     const void * pData )
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:84]: Add return statement.

**Review by gshvang — APPROVED** [review]

**Review by gshvang — DISMISSED** [review]

**Review by gshvang — DISMISSED** [review]

File: test/cbmc/proofs/processValidFileContext/processValidFileContext_harness.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    /* Initialize otaAgent. */
+    otaInterface.pal.getPlatformImageState = getPlatformImageStateStub;
+    otaInterface.pal.reset = resetPalStub;
+    otaInterface.os.event.send = sendEventStub;
+
+    /* otaAgent.pOtaInterface can never be NULL as it is checked during the start of OTA Agent
+     * in the receiveAndProcessOtaEvent function.*/
+    otaAgent.pOtaInterface = &otaInterface;
+    otaAgent.fileContext = fileContext;
```
Comment (paulbartell) [inline test/cbmc/proofs/processValidFileContext/processValidFileContext_harness.c:53]: the remainder of the otaAgent should be unconstrained.

File: test/cbmc/proofs/processValidFileContext/processValidFileContext_harness.c [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    err = processValidFileContext();
+
+    __CPROVER_assert( ( err >= OtaErrNone ) && ( err <= OtaErrActivateFailed ),
+                      "Error: Return value from processValidFileContext should follow values of OtaErr_t enum." );
```
Comment (paulbartell) [inline test/cbmc/proofs/processValidFileContext/processValidFileContext_harness.c:58]: The resulting eventMsg.eventId, otaAgent.fileContext, and ImageState/reason fields should be validated.

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**